kuma/core/middleware.py

```diff
@@ -35,7 +36,10 @@ def process_request(self, request):
             new_path = prefixer.fix(prefixer.shortened_path)
             query = dict((smart_str(k), v) for
                          k, v in request.GET.iteritems() if k != 'lang')
-            return HttpResponsePermanentRedirect(urlparams(new_path, **query))
+
+            # Never use HttpResponsePermanentRedirect here.
+            # Its a temporary redirect and should return with http 302, not 301
```

Comment:
> `https://developer.mozilla.org/en-US/docs/Mozilla?lang=de`
> 
> should be a permanent redirect to:
> 
> `https://developer.mozilla.org/de/docs/Mozilla`

Like, If I go to `bn-BD` locale for some reason, I dont want to be there permanently.